tests/deploy/sa-agent-pytest.yml

```diff
@@ -483,21 +511,22 @@ spec:
       labels:
         app: elastic-agent
     spec:
-      # Tolerations are needed to run Elastic Agent on Kubernetes master nodes.
-      # Agents running on master nodes collect metrics from the control plane components (scheduler, controller manager) of Kubernetes
+      # Tolerations are needed to run Elastic Agent on Kubernetes control-plane nodes.
+      # Agents running on control-plane nodes collect metrics from the control plane components (scheduler, controller manager) of Kubernetes
       tolerations:
+        - key: node-role.kubernetes.io/control-plane
+          effect: NoSchedule
         - key: node-role.kubernetes.io/master
           effect: NoSchedule
       serviceAccountName: elastic-agent
       hostNetwork: true
       dnsPolicy: ClusterFirstWithHostNet
       containers:
         - name: elastic-agent
-          image: 704479110758.dkr.ecr.eu-west-1.amazonaws.com/elastic-agent:8.6.0-SNAPSHOT
+          image: docker.elastic.co/beats/elastic-agent:8.6.0-SNAPSHOT
```

Comment:
I though there was a reason behind us not using the public agent image?

Reply:
it's just a name. In kind deployment all images are loaded locally. This will serve us in the future when we will need to deploy agent on EKS.